lib/ansible/modules/cloud/scaleway/scaleway_compute.py

```diff
@@ -271,30 +291,33 @@ def public_ip_payload(compute_api, public_ip):
 
 def create_server(compute_api, server):
     compute_api.module.debug("Starting a create_server")
-    target_server = None
-    payload = {"enable_ipv6": server["enable_ipv6"],
-               "tags": server["tags"],
-               "commercial_type": server["commercial_type"],
-               "image": server["image"],
-               "dynamic_ip_required": server["dynamic_ip_required"],
-               "name": server["name"],
-               "organization": server["organization"]}
-
-    response = compute_api.post(path="servers",
-                                data=payload)
+    created_server = None
+    data = {"enable_ipv6": server["enable_ipv6"],
+            "tags": server["tags"],
+            "commercial_type": server["commercial_type"],
+            "image": server["image"],
+            "dynamic_ip_required": server["dynamic_ip_required"],
+            "name": server["name"],
+            "organization": server["organization"]
+            }
+
+    if server["security_group"]:
+        data["security_group"] = server["security_group"]
+
+    response = compute_api.post(path="servers", data=data)
 
     if not response.ok:
         msg = 'Error during server creation: (%s) %s' % (response.status_code, response.json)
         compute_api.module.fail_json(msg=msg)
 
     try:
-        target_server = response.json["server"]
+        created_server = response.json["server"]
```

Comment:
Is the refactoring of the name necessary? It adds a lot of lines in the review.

Reply:
I added the [explantation](https://github.com/ansible/ansible/pull/45699#discussion_r221433211)